Attribute Analysis.ipynb

```diff
@@ -602,6 +606,379 @@
+    "        r = [r1]\n",
+    "        for i in range(1, num_attrs+1):\n",
+    "            r.append([x + barWidth for x in r[len(r)-1]])\n",
+    "\n",
+    "        category_attr_ratios = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "        max_categories = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "        for category in range(len(supercategory_counts[0])):\n",
+    "            max_category = 0\n",
+    "            min_category = 1\n",
+    "            max_attr = -1\n",
+    "            for attr in range(num_attrs):\n",
+    "                if supercategory_counts[attr][category] > max_category:\n",
+    "                    max_category = supercategory_counts[attr][category]\n",
+    "                    max_attr = attr\n",
+    "                if supercategory_counts[attr][category] < min_category:\n",
+    "                    min_category = supercategory_counts[attr][category]\n",
+    "            category_attr_ratios[category] = max_category / min_category\n",
+    "            max_categories[category] = max_attr\n",
+    "            \n",
+    "        order = np.argsort(category_attr_ratios)\n",
+    "        r = np.array(r)\n",
+    "        for i in range(num_attrs-1, -1, -1):\n",
+    "            plt.barh(r[i], supercategory_counts[i][order], height=barWidth, color=COLORS[i], edgecolor='white', label=attr_names[i])\n",
+    "        \n",
+    "        ticks = [r[0]+(num_attrs/2)*barWidth for i in range(len(r1))][0]\n",
+    "        plt.yticks(ticks, np.array([datasets.DEFAULT_GROUPINGS_TO_NAMES[i+1] for i in range(len(r1))])[order], fontsize=fontsize)\n",
+    "        plt.xticks(fontsize=fontsize)\n",
+    "        plt.legend(loc='best', fontsize=fontsize)\n",
+    "        plt.ylabel('Object Category', fontsize=fontsize, labelpad=20)\n",
+    "        plt.xlabel('Fraction of Labelled Images that contain this Category', fontsize=fontsize, labelpad=20, x=.3)\n",
+    "        plt.tight_layout()\n",
+    "        plt.gcf().subplots_adjust(bottom=0.18)\n",
+    "        plt.gcf().subplots_adjust(left=0.28)\n",
+    "        if first_pass:\n",
+    "            to_write[3] = ['(M2) Distribution of object categories each attribute appears with, sorted by ratio between the max count attribute and min count attribute.']\n",
+    "            plt.savefig(\"results/{0}/{1}/1.png\".format(folder_name, save_loc))\n",
+    "        plt.show()\n",
+    "\n",
+    "# Graphs the ratio of instance counts if they are statistically significant\n",
+    "indices_to_keep = [i for i in range(len(instance_counts[0])) if categories[i] not in dataset.people_labels]\n",
+    "#pspecific_indices = [[i for i in range(ind*(num_attrs**2),ind*(num_attrs**2)+(num_attrs**2))] for ind in indices_to_keep]\n",
+    "pspecific_indices = []\n",
+    "values_forobject = 0\n",
+    "for i in range(num_attrs):\n",
+    "    for a in range(i+1, num_attrs):\n",
+    "        values_forobject += 1\n",
```

Comment:
pretty sure this whole for loop could be replaced by values_forobject = (num_attrs)*(num_attrs+1)/2 

please verify